buildbot/configure.py

```diff
@@ -74,7 +74,8 @@ def do_configure(args):
         "-DLLVM_ENABLE_DOXYGEN={}".format(llvm_enable_doxygen),
         "-DLLVM_ENABLE_SPHINX={}".format(llvm_enable_sphinx),
         "-DBUILD_SHARED_LIBS={}".format(llvm_build_shared_libs),
-        "-DSYCL_ENABLE_XPTI_TRACING=ON" # Explicitly turn on XPTI tracing
+        "-DSYCL_ENABLE_XPTI_TRACING=ON", # Explicitly turn on XPTI tracing
+        "-DSYCL_USE_SANITIZER=ON"
```

Comment:
IIRC, sanitation brings some overhead and using this script is recommended in get started guide, which making "ON" a default value for almost everyone who is building the library from sources.

I think we should have two separate configuration modes for the project:
- regular (user mode) - assuming that CI status is "green" just build me a SYCL compiler toolchain I can use in my daily development.
- CI (developers mode) - enable as much checks as possible

I think it's more reasonable to use "regular" mode by default, so that users following get started guide won't be surprised with unexpected overhead or checks.
It should be easy for developers to enable "CI" mode as well.

One of the possible implementations can be:
- we add `-ci` option to buildbot scripts, which is `off` by default and set by CI checks and can be used by developers for the development and/or to reproduce CI behavior.

@tfzhu, does it sounds reasonable to you? If so we can add this option ASAP and gradually separate "user" and "developer" modes.

"Enable -Werror flag" is another option in addition to sanitizer, which should be disabled in "user" mode. Users have different C++ toolchains on their systems and we have a number of bug reports about failed builds due to new warnings.

Reply:
I'm OK with adding `-ci` option